### ISSUE TITLE
mean reward suddenly drops to -1 and model wont learn anymore

### ISSUE THREAD

**Issue by chenh1001** [Unity-Technologies/ml-agents#2059]
i am trying to train an agent to play basket ball (just shoot to a basket). 
as a reward i am giving the model +1 when a basket is made and -1 when out of bounds.
the model is learning and even getting to a super human level (mean reward at 0.9) but suddenly the mean reward drops to -1 and stays there, not learning anything at all. i dont know why this is happening and im preaty stuck :/. i have also tried changing most of the hyperparameters but nothing worked. here are they:
trainer: ppo
    batch_size: 512
    beta: 5.0e-3
    buffer_size: 2048
    epsilon: 0.1
    gamma: 0.995
    hidden_units: 32
    lambd: 0.95
    learning_rate: 8.0e-4
    max_steps: 4e6
    memory_size: 512
    normalize: true
    num_epoch: 3
    num_layers: 2
    time_horizon: 64
    sequence_length: 64
    summary_freq: 1000
    use_recurrent: false
    use_curiosity: false
    curiosity_strength: 0.01
    curiosity_enc_size: 128

**Comment by roboserg** [comment]
did you try train longer after it drops to -1 ? Like 100k steps, longer?
Also try default hyperparameters (learning rate of 3.0e-4). Also try 
    batch_size: 10
    buffer_size: 100

**Comment by chenh1001** [comment]
thank for replying! actually i have allready figured it out, the problem was that the max steps value was too high and reward dropped to -1 after the model couldnt learn more for a long period so i made max steps smaller and now the results and great!

**Comment by roboserg** [comment]
wait, do you call Done() after you score a goal or miss? I don't understand how long max steps would cause your problem

**Comment by chenh1001** [comment]
yes i do, im coling Done() if a score is made or missed.

**Comment by roboserg** [comment]
Then I don't understand why the reward would drop to -1. It would mean your model would miss constantly.

**Comment by chenh1001** [comment]
after succeeding for a long period the reward would have just droped instantly and model would allways miss:/
